lib/library_main.rb

```diff
@@ -0,0 +1,42 @@
+require 'yaml'
+require 'date'
+class Library_main
+
+    require 'date'
+    attr_accessor :collection
+
+    def initialize
+      @collection = YAML.load_file('./lib/data.yml')
+    end
+
+    def check_avl (title,author)
+      book = @collection.detect { |obj| obj[:item][:title] == title && obj[:item][:author] == author } 
+      return book[:available]
+    end
+
+    def check_out (title,author)
+      avl = check_avl(title,author)
+      if avl == true 
+        index = @collection.find_index { |obj| obj[:item][:title] == title && obj[:item][:author] == author } 
+        @collection[index][:available] = false
+        @collection[index][:return_date] = Date.today.next_day(30).strftime('%d/%m/%y')
+        #@collection[index][:return_date] = Date.today + 30
+        File.open('./lib/data.yml', 'w') { |f| f.write @collection.to_yaml }
+      end
+    end
+     
+    def check_in (title,author)
+      index = @collection.find_index { |obj| obj[:item][:title] == title && obj[:item][:author] == author } 
```

Comment:
avoid the variable name `index` in the future as it has specific meaning when we are dealing with frameworks